tests/rest/admin/test_user.py

```diff
@@ -320,6 +324,55 @@ def nonce():
         self.assertEqual(400, int(channel.result["code"]), msg=channel.result["body"])
         self.assertEqual("Invalid user type", channel.json_body["error"])
 
+    def test_register_mau_limit_reached(self):
+        """
+        Register user with Admin API if MAU limit is reached.
+        """
+        handler = self.hs.get_registration_handler()
+        store = self.hs.get_datastore()
+        auth_blocking = self.hs.get_auth()._auth_blocking
+
+        # Configure MAU limit
+        auth_blocking._limit_usage_by_mau = True
+        auth_blocking._max_mau_value = 2
+        auth_blocking._mau_trial_days = 0
```

Comment:
better to do this with `@override_config`, rather than poking into the guts of the auth_blocking object

or better yet: override `default_config`.

See also the docstring on `HomeserverTestCase`.